qiskit_dynamics/solver_classes.py

```diff
@@ -0,0 +1,335 @@
+                             :meth:`LindbladModel.set_evaluation_mode`
+                             (if dissipators in model) for valid modes.
+            rwa_cutoff_freq: Rotating wave approximation cutoff frequency. If ``None``, no
+                             approximation is made.
+        """
+
+        model = None
+        if dissipator_operators is None:
+            model = HamiltonianModel(
+                operators=hamiltonian_operators,
+                signals=hamiltonian_signals,
+                rotating_frame=rotating_frame,
+                drift=drift,
+                evaluation_mode=evaluation_mode,
+            )
+            self._signals = hamiltonian_signals
+        else:
+            model = LindbladModel(
+                hamiltonian_operators=hamiltonian_operators,
+                hamiltonian_signals=hamiltonian_signals,
+                dissipator_operators=dissipator_operators,
+                dissipator_signals=dissipator_signals,
+                rotating_frame=rotating_frame,
+                drift=drift,
+                evaluation_mode=evaluation_mode,
+            )
+            self._signals = (hamiltonian_signals, dissipator_signals)
+
+        self._rwa_signal_map = None
+        if rwa_cutoff_freq is not None:
+            model, rwa_signal_map = rotating_wave_approximation(
+                model, rwa_cutoff_freq, return_signal_map=True
+            )
+            self._rwa_signal_map = rwa_signal_map
+
+        self._model = model
+
+    @property
+    def rotating_frame(self) -> RotatingFrame:
+        """Rotating frame of the model."""
+        return self.model.rotating_frame
+
+    @rotating_frame.setter
+    def rotating_frame(self, new_rotating_frame: Union[Array, RotatingFrame]):
+        """Set rotating frame for the model."""
+        self.model.rotating_frame = new_rotating_frame
+
+    @property
+    def model(self) -> BaseGeneratorModel:
+        """The model."""
+        return self._model
+
+    @property
+    def signals(self) -> SignalList:
+        """Solver signals."""
+        return self._signals
+
+    @signals.setter
+    def signals(self, new_signals: Union[List[Signal], SignalList]):
+        """Set signals for the solver, and pass to the model."""
+        self._signals = new_signals
+        if self._rwa_signal_map is not None:
+            new_signals = self._rwa_signal_map(new_signals)
+        self.model.signals = new_signals
+
+    @property
+    def evaluation_mode(self) -> str:
+        """Evaluation mode for the model."""
+        return self.model.evaluation_mode
+
+    def set_evaluation_mode(self, new_mode: str):
+        """Set model evaluation mode. See documentation for
+        :class:`~qiskit_dynamics.models.HamiltonianModel`
+        or :class:`~qiskit_dynamics.models.LindbladModel`
+        (if dissipators present in model) for valid methods.
+        """
+        self.model.set_evaluation_mode(new_mode)
+
+    def copy(self) -> "Solver":
```

Comment:
The main reason I put this method is that when using JAX, copying the `Solver` (or model if working with that directly) is often a necessary step in building pure functions which can be JAX-transformed safely. I thought at some point we can mayb emake the `.copy()` function smarter so that it isn't just doing a `deepcopy`, but I guess until then we can just get rid of it?